Extensions/AboutUsParser.js

```diff
@@ -0,0 +1,267 @@
+ * @param {String} language
+ * @param {Boolean} append_file Whether to append or overwrite already existing file 
+ */
+async function createComponent(Fpath = null, name, string, language, append_file = true, fileName = 'school') {
+    Fpath = Fpath ? Fpath : C.path;
+    const component = `:^) ${name} :::`;
+    let path = Path.join(Fpath, language, `${fileName}.html`);
+    writeFile(path, `${component}\r\n${string}`, append_file);
+    console.log(`Successfully created component ${component},\nAppend File Mode: ${append_file}`);
+}
+
+/** Asynchronously writes a string to a file
+ * 
+ * @param {String} path Path of the file
+ * @param {String} string String to write into the file
+ * @param {Boolean} appendFile 
+ * @returns {void}
+ */
+function writeFile(path, string, appendFile) {
+    try {
+        switch (appendFile) {
+            case true:
+                fs.appendFile(path, `${string}\r\n`, { flag: 'a+', mode: 0666 }, (err) => {
+                    if (err) throw new Error(err);
+                });
+                break;
+            case false:
+                fs.writeFile(path, `${string}\r\n`, { flag: 'w+', mode: 0666 }, (err) => {
+                    if (err) throw new Error(err);
+                });
+                break;
+            default:
+                throw new Error('No valid action was chosen for writing a file!');
+        }
+    } catch (error) {
+        console.error(error)
+        return;
+    }
+}
+
+/** Creates a Json file with our data
+ * 
+ * @param {Array} array Array of paragraph objects
+ */
+function saveAsJson(array) {
+    fs.writeFileSync(C.dPath + C.jsonFile, JSON.stringify(array, null, 4), {flag: 'w+', mode: 0666});
+}
+
+/** Loads a Json file and returns its content decoded
+ * 
+ * @param {String} file Path to the Json file 
+ * @returns {String}
+ */
+function loadFromJson(file) {
+    return JSON.parse(fs.readFileSync(file, { mode: 0666 }));
+}
+
+/** Creates components files from array
+ * 
+ * @param {Array} array 
+ */
+function arrayIntoComponents(array) {
+    array.map((section, sectionIndex) => {
+        section.content.map(content => {
+            //console.log(content);
+            let string = getStringFromArray(content.data);
+            if(sectionIndex == 0)
+                createComponent(C.path, section.folder, string, content.language, false, 'about');
+            else
+                createComponent(C.path, section.folder, string, content.language, true, 'about');
+        });
+    });
+}
+
+/**
+ * 
+ * @param {Array} array 
+ * @deprecated This will be added later on, it should check whether Json file contains objects with paths to photos and ask whether to combine them with newly parsed objects array of data
+ */
+function checkJsonForPhotos(parsedJson) {
+    let arr = parsedJson;
+
+    return parsedJson;
+}
+
+Files(C.dPath, async res => {
+    var arr = await createFilesArray(res);
+    var folders = getFolders(C.dPath);
```

Comment:
why are some thing `var` and some `let`?

Reply:
Depends on the scope they are used in.